### DISCUSSION TITLE
FIDO Alliance tests

### DISCUSSION THREAD

**Discussion by ranierimazili** [webauthn-open-source/fido2-lib#173] [General]
Hello,

The Open Finance Brasil (the biggest Open Finance ecosystem in the world) has approved the usage of FIDO protocol to authorize payments directly on wallet.
Google Wallet is the first use case already approved and you can see a demo on the GIF below:

![google_wallet_ofb](https://github.com/user-attachments/assets/1be42713-fba5-495e-9c6b-65c2f056b648)

But, the banks can use only FIDO Alliances approved providers.
The only open source provider that's already approved I could see is [line server](https://github.com/line/line-fido2-server) and it's a Java alternative.

To pass the FIDO tests, is necessary to create a REST project, providing some API's that uses the fido server.
I'm thinking on start this project, but first would like to see if anyone has started it before to not duplicate the efforts.

Thanks

**Comment by JamesCullum** [comment]
That sound's like a great idea! Maybe there is an OpenAPI spec that we can use, and implement via our backend?

**Reply by ranierimazili** [reply]
Yes, there's: https://github.com/fido-alliance/conformance-test-tools-resources/blob/main/docs/FIDO2/Server/Conformance-Test-API.md#registration

I started the implementation and I just passed the the first block (of total of 6 blocks).

I'll need some help to solve protencial bugs that maybe the tests will find because unfortunatelly I still don't understand the FIDO protocol so deply to solve it myself.
For each problem, do you prefer that I create a new issue or can I post all of them directly on this thread?

In a future comment (maybe today) on this thread I'll post the instructions to perform the tests and the repo where I'm building the conformance suite to fido2-lib.

**Reply by JamesCullum** [reply]
Maybe it's the easiest to simply do a fork where you implement this, and we link to it? Depending on how much overhead we need to make this testable for the compliance. 

Then we can do the tracking in the issue tracker of the fork.
